icu4c/source/test/cintltst/ccapitst.c

```diff
@@ -3035,7 +3035,7 @@ static void TestJ1968(void) {
 
     err = U_ZERO_ERROR;
     myConvName[UCNV_MAX_CONVERTER_NAME_LENGTH-1] = ',';
-    strncpy(myConvName + UCNV_MAX_CONVERTER_NAME_LENGTH, "locale=", 7);
+    memcpy(myConvName + UCNV_MAX_CONVERTER_NAME_LENGTH, "locale=", 7);
```

Comment:
this isn't library code, but do we want to use a #define to document the use of memcpy-for-string here?

Reply:
nah, we use memcpy for strings in lots of places already

however, i think you need to copy 8 bytes in order to get myConvName NUL-terminated

Reply:
yes, we do use memcpy.  The idea was to use a macro just to document the usage. But it's minor, especially in test code.

Reply:
The gcc warning for this one was that the string is 8 bytes long (including NUL), but strncpy is limited to copy only 7 bytes.  Changing strncpy to memcpy is equivalent and silences the warning.